src/Serilog/Parsing/MessageTemplateParser.cs

```diff
@@ -86,9 +86,8 @@ static MessageTemplateToken ParsePropertyToken(int startAt, string messageTempla
             next = startAt + 1;
 
             var rawText = messageTemplate.Substring(first, next - first);
-            var tagContent = messageTemplate.Substring(first + 1, next - (first + 2));
-            if (tagContent.Length == 0 ||
-                !IsValidInPropertyTag(tagContent[0]))
```

Comment:
`!IsValidInPropertyTag(tagContent[0])` performs the check on `T` in `{Test}` but all characters in `{Test}` have been already checked by
`while (startAt < messageTemplate.Length && IsValidInPropertyTag(messageTemplate[startAt]))`